Code/Framework/AzFramework/AzFramework/Physics/Material/PhysicsMaterialManager.cpp

```diff
@@ -62,24 +62,27 @@ namespace Physics
             return materialFound;
         }
 
-        // Take a reference so we can mutate it.
-        AZ::Data::Asset<AZ::Data::AssetData> assetLocal = materialAsset;
-        if (!assetLocal.IsReady())
+        auto newMaterial = CreateMaterialInternal(id, materialAsset);
+        if (newMaterial)
         {
-            assetLocal.QueueLoad();
-            assetLocal.BlockUntilLoadComplete();
-
-            // Failed to load the asset
-            if (!assetLocal.IsReady())
-            {
-                return nullptr;
-            }
+            m_materials.emplace(newMaterial->GetId(), newMaterial);
         }
 
-        auto newMaterial = CreateMaterialInternal(id, assetLocal);
-        if (newMaterial)
+        // Block until the material asset is loaded.
+        // It's important that the loading of physics material assets is not delayed,
+        // otherwise objects would react differently during simulation for a few
+        // frames until the material is loaded, causing unexpected behaviors.
+        AZ::Data::Asset<AZ::Data::AssetData> assetLocal = materialAsset; // Take a reference so we can mutate it for loading.
+        if (!assetLocal.IsReady())
         {
-            m_materials.emplace(newMaterial->GetId(), newMaterial);
+            assetLocal.QueueLoad();
+
+            // The call to BlockUntilLoadComplete must happen after the new material
+            // has been added to m_materials container. This is because BlockUntilLoadComplete
+            // will dispatch events for other assets while waiting, and if other OnAssetRead
+            // code ends up trying find/create the same material it needs to find it in the
+            // container and return it.
+            assetLocal.BlockUntilLoadComplete();
```

Comment:
An alternative approach to this would be to make all the physx components that use physx materials to delay their initialization until all the physics materials it uses are loaded. But we cannot go for that solution at the moment because all the colliders need to create the physx shapes during activation. The current implementation cannot wait a few frames later to create the shapes, the rigid body components also need the shapes to create the simulated body on "entity activated" event.

If we change how of all physx components are initialized with a more dynamic approach based on events to allow the creation of shapes and simulated bodies at any time, then it would be possible to delay the creation until all the assets are ready.

Not sure if it's a good idea overall to delay the creation of physical entities though, that might lead to objects falling though floors, or trespassing walls.